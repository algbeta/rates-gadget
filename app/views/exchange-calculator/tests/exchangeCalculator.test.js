import React from 'react';
import renderer from 'react-test-renderer';
import ExchangeCalculator from '../exchangeCalculator';

jest.mock('../exchangeCard', () => () => <div className="exchangeCard" />);
jest.mock('../../spinner', () => () => <div className="spinner" />);

describe('ExchangeCalculator tests', () => {
  test('ExchangeCalculator renders correctly', () => {
    const tree = renderer
      .create(
        <ExchangeCalculator
          fetchCurrenciesPeriodically={() => {}}
          transfer={() => {}}
          amount={150}
          from="from"
          to="to"
          validationFailed={false}
          sameCurrency={false}
          areCurrenciesLoaded
          transactionError=""
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ExchangeCalculator with transaction error', () => {
    const tree = renderer
      .create(
        <ExchangeCalculator
          fetchCurrenciesPeriodically={() => {}}
          transfer={() => {}}
          amount={150}
          from="from"
          to="to"
          validationFailed={false}
          sameCurrency={false}
          areCurrenciesLoaded
          transactionError="Something went wrong!"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("ExchangeCalculator renders spinner when currencies aren't loaded", () => {
    const tree = renderer
      .create(
        <ExchangeCalculator
          fetchCurrenciesPeriodically={() => {}}
          transfer={() => {}}
          amount={150}
          from="from"
          to="to"
          validationFailed={false}
          sameCurrency={false}
          areCurrenciesLoaded={false}
          transactionError=""
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ExchangeCalculator method tests', () => {
  test('componentDidMount', () => {
    const context = {
      props: {
        fetchCurrenciesPeriodically: jest.fn(),
      },
    };

    ExchangeCalculator.prototype.componentDidMount.call(context);
    expect(context.props.fetchCurrenciesPeriodically).toHaveBeenCalledWith(
      true,
    );
  });

  test('componentWillUnmount', () => {
    const context = {
      props: {
        fetchCurrenciesPeriodically: jest.fn(),
      },
    };

    ExchangeCalculator.prototype.componentWillUnmount.call(context);
    expect(context.props.fetchCurrenciesPeriodically).toHaveBeenCalledWith(
      false,
    );
  });
});
