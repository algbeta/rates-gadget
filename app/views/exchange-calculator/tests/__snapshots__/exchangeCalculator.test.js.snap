// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeCalculator tests ExchangeCalculator renders correctly 1`] = `
<div
  className="container"
>
  <h1>
    Currency exchange
  </h1>
  
  <div
    className="exchangeCard"
  />
  <div
    className="exchangeCard"
  />
  <button
    className="btn btn-primary float-right"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Transfer
  </button>
</div>
`;

exports[`ExchangeCalculator tests ExchangeCalculator renders spinner when currencies aren't loaded 1`] = `
<div
  className="spinner"
/>
`;

exports[`ExchangeCalculator tests ExchangeCalculator with transaction error 1`] = `
<div
  className="container"
>
  <h1>
    Currency exchange
  </h1>
  <div
    className="alert-secondary"
    role="alert"
  >
    Something went wrong!
  </div>
  <div
    className="exchangeCard"
  />
  <div
    className="exchangeCard"
  />
  <button
    className="btn btn-primary float-right"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Transfer
  </button>
</div>
`;
